faker/providers/internet/__init__.py

```diff
@@ -178,6 +210,45 @@ def ipv4(self, network=False):
             address = str(ip_network(address, strict=False))
         return address
 
+    def ipv4_network_class(self):
+        return self.random_element('abc')
+
+    def ipv4_private(self, address_class=None, network=False):
+        address_class = address_class or self.ipv4_network_class()
+        min_, max_, netmask = IPV4_NET_CLASSES[address_class]
+        address = str(ip_address(
+            self.generator.random.randint(min_, max_)))
+        if network:
+            address += '/' + str(self.generator.random.randint(netmask, 31))
+            address = str(ip_network(address, strict=False))
+        return address
+
+    def ipv4_public(self, network=False, address_class=None):
+        address_class = address_class or self.ipv4_network_class()
+        public_net = IPV4_PUBLIC_NETS[address_class]
+        private_blocks = IPV4_PRIVATE_NET_BLOCKS[address_class]
+        # Make valid IP ranges
+        net_ranges = []
+        if public_net[0] != private_blocks[0][0]:
```

Comment:
Could you explain why do we need this check? This code is quite hard to follow, comments would help

Reply:
I check if the 1st available range starts after the 1st private block.
Example:
Class C: 192.0.0.0/24
The first address 192.0.0.0 is in 192.0.0.0/8 which is private.
So the 1st `net_ranges` will be from 192.0.0.8 to 192.X.X.X

I'll add comments